src/renderer/navigation/url-param.ts

```diff
@@ -0,0 +1,115 @@
+// Manage observable URL-param via location.search
+import { IObservableHistory } from "mobx-observable-history";
+
+export interface UrlParamInit<V = any> {
+  name: string;
+  isSystem?: boolean;
+  defaultValue?: V;
+  multiValues?: boolean; // false == by default
+  multiValueSep?: string; // joining multiple values with separator, default: ","
+  skipEmpty?: boolean; // skip empty value(s), e.g. "?param=", default: true
+  parse?(values: string[]): V; // deserialize from URL
+  stringify?(values: V): string | string[]; // serialize params to URL
+}
+
+export class UrlParam<V = any | any[]> {
+  static SYSTEM_PREFIX = "lens-";
+
+  public name: string;
+  public urlName: string;
+
+  constructor(private init: UrlParamInit<V>, private history: IObservableHistory) {
+    const { isSystem, name, skipEmpty = true } = init;
+
+    this.name = name;
+    this.init.skipEmpty = skipEmpty;
+
+    // prefixing to avoid collisions with extensions
+    this.urlName = `${isSystem ? UrlParam.SYSTEM_PREFIX : ""}${name}`;
```

Comment:
these should be getters

Reply:
Doesn't really matter, probably better to mark with `readonly`